low-level/sensors/impl/generic/node_hw.py

```diff
@@ -243,34 +257,29 @@ def initialize(self, conf_reader, msgQlist, products):
         self._bmc_user = Conf.get(GLOBAL_CONF, BMC_USER_KEY, 'ADMIN')
         _bmc_secret = Conf.get(GLOBAL_CONF, BMC_SECRET_KEY, 'ADMIN')
         self._bmc_ip = Conf.get(GLOBAL_CONF, BMC_IP_KEY, '')
-        self._channel_interface = Conf.get(SSPL_CONF,
-            f"{self.BMC_INTERFACE}>{self.BMC_CHANNEL_IF}", 'system')
+
+        self._channel_interface = Conf.get(
+            SSPL_CONF, f"{BMC_INTERFACE}>{BMC_CHANNEL_IF}", 'system')
+        self.active_bmc_if = None
+        self.get_interface_from_cache()
```

Comment:
what is this interface from cache? Why the need to cache here

Reply:
>>what is this interface from cache? 
Reading the data from persistent cache.

>>Why the need to cache here
In case of lan fault we switched to default kcs interface(this we have decided in cortx-v1) 
once switched to kcs interface, we still  check for fault-resolved alert for lan.
if we did not use persistent cache here then in case of sspl restart we will loose the lan fault data,
hence for handling this case I have used persistent cache here.

Reply:
ok, add these comments around relevant code

Reply:
got it, add comments around relevant code

Reply:
Done.

Reply:
add comments on why part, what we can understand from code

Reply:
Added detailed comment inside get_interface_from_cache function, also updated comment here.